tmk_core/common/timer.h

```diff
@@ -47,6 +48,25 @@ uint32_t timer_elapsed32(uint32_t last);
 #define timer_expired(current, future) ((uint16_t)(current - future) < UINT16_MAX / 2)
 #define timer_expired32(current, future) ((uint32_t)(current - future) < UINT32_MAX / 2)
 
+// Use an appropriate timer integer size based on architecture (16-bit will overflow sooner)
+#if defined(__AVR__)
```

Comment:
There really shouldn't be the introduction of platform ifdefs here, given this file is mean to be the abstraction over timers. Be it that its swapped to duck typing the platform as 32bit or a define to indicate such, set somewhere else

Reply:
Is this ok?
```
#if FAST_TIMER_SIZE < 32
#    define TIMER_DIFF_FAST(a, b) TIMER_DIFF_16(a, b)
#    define timer_expired_fast(current, future) timer_expired(current, future)
typedef uint16_t fast_timer_t;
fast_timer_t inline timer_read_fast(void) { return timer_read(); }
fast_timer_t inline timer_elapsed_fast(fast_timer_t last) { return timer_elapsed(last); }
#else
#    define TIMER_DIFF_FAST(a, b) TIMER_DIFF_32(a, b)
#    define timer_expired_fast(current, future) timer_expired32(current, future)
typedef uint32_t fast_timer_t;
fast_timer_t inline timer_read_fast(void) { return timer_read32(); }
fast_timer_t inline timer_elapsed_fast(fast_timer_t last) { return timer_elapsed32(last); }
#endif
```

I tried using `sizeof(fast_timer_t) == 1` but it just becomes harder to read.

I initially put the `typedef` in a platform header file but I think it's better in `timer.h` to ensure it's consistent with one of the two types available from the timer API.